ui/components/component-library/popover/popover.test.tsx

```diff
@@ -0,0 +1,201 @@
+/* eslint-disable jest/require-top-level-describe */
+import { render, fireEvent } from '@testing-library/react';
+import React from 'react';
+import { PopoverHeader } from '../popover-header';
+import { Popover } from './popover';
+import { PopoverPosition } from './popover.types';
+
+describe('Popover', () => {
+  it('should render popover element correctly', () => {
+    const { getByTestId, getByText, container } = render(
+      <Popover data-testid="popover" isOpen={true} isPortal={false}>
+        Popover
+      </Popover>,
+    );
+    expect(getByText('Popover')).toBeDefined();
+    expect(container.querySelector('popover')).toBeDefined();
+    expect(getByTestId('popover')).toHaveClass('mm-popover');
+    expect(container).toMatchSnapshot();
+  });
+
+  it('should render popover children', () => {
+    const { getByText } = render(
+      <Popover isOpen={true}>Popover content goes here</Popover>,
+    );
+    expect(getByText('Popover content goes here')).toBeDefined();
+  });
+
+  it('should render popover back button', () => {
+    const onBackTest = jest.fn();
+    const { getByTestId } = render(
+      <Popover data-testid="popover" isOpen={true}>
+        <PopoverHeader
+          onBack={onBackTest}
+          backButtonProps={{ 'data-testid': 'back' }}
+        >
+          onClose Test
+        </PopoverHeader>
+      </Popover>,
+    );
+
+    const backButton = getByTestId('back');
+    fireEvent.click(backButton);
+
+    expect(onBackTest).toHaveBeenCalled();
+  });
```

Comment:
I think this test could just be in the `PopoverHeader` component? There doesn't seem to be anything specific to the `Popover` in this test?